### DISCUSSION TITLE
Allow FPS field to be changed

### DISCUSSION THREAD

**Discussion by srnyx** [maruohon/minihud#180] [Ideas]
Similar to how the coordinated info toggle can be changed, allow the FPS info toggle to be edited.
Right now it's `X fps`, but I'd rather have `X FPS` or `FPS: X`

**Comment by maruohon** [comment]
All of the info lines will be fully configurable in future versions. The info lines will basically become configurable strings, that can contain format specifiers or substitutions, whatever you would call them, to include the bits of dynamic information.

So for example the default FPS info line string would be something like:
`{FPS} fps`

Then you can put any text you want there formatted in any way you want, and the `{FPS}` part would get replaced by the actual number.

**Reply by srnyx** [reply]
> All of the info lines will be fully configurable in future versions. The info lines will basically become configurable strings, that can contain format specifiers or substitutions, whatever you would call them, to include the bits of dynamic information.
> 
> So for example the default FPS info line string would be something like: `{FPS} fps`
> 
> Then you can put any text you want there formatted in any way you want, and the `{FPS}` part would get replaced by the actual number.

Would that also allow for things like `{FPS} FPS, {PING} ms` in the same line? Would be cool if it was basically just like a text document that you could edit in-game and had all of the variables listed on the side (and in a comment in the file itself at the top). A new line in the editor would mean a new line in the actual thing.
You could also add a way to create multiple "instances" (not sure what to call them) that could be moved around separately and also be able to resuse variables (so I could have it say FPS in the top left and ping in the bottom right).
Here's what I'm imagining:
![image](https://user-images.githubusercontent.com/25808801/140845359-3924d92d-da64-40e2-bc75-306e78077565.png)

This is a very extreme idea and probably really hard to implement, but I think it'd satisfy a ton of people.

**Reply by maruohon** [reply]
Yes since it's going to just substitute the placeholder with the value, you can put whatever and however many things you want on the same line. The current info lines will probably be provided as some kind of default "templates" or pre-existing configured info lines. I'm not sure yet what the editor will look like exactly, but you will have some kind of list of the existing info things/placeholders to choose from to add to the lines.

I'm not sure if it will allow a multi-line config, the intention was that you would add different lines as different entries, and you can then also drag-order them in some kind of a list. I don't know what a line-split one multi-line config would offer over that? Except that the lines will be keybindable, so I guess it would allow toggling multiple lines with a single keybind a bit "easier", although you could also just bind the same key to multiple separate line instances. Maybe it might be worth also supporting a multi-line entry? Although it would complicate/messy up the drag-ordering list and other things that may provide a preview or the contents in some kind of a list entry... so I guess it's more likely not to happen, and you would just add separate lines.

Edit: The new malilib Info Overlay and Info Area stuff and how MiniHUD will hook into them should also allow you to put multiple separate MiniHUD info line things in different screen corners. That underlying Info Overlay and Info Area system already exists and works in the 1.12.2 development code, but the part where MiniHUD could handle and configure and present multiple separate info areas does not exist yet.
